### PR TITLE
perf(sql): reduce unnecessary TableReader partition reloads

### PR DESCRIPTION
**Pull request by ideoma**
Reduce performance hit by re-opening table partitions

- generously bump table reader max open partitions to 10,000 from 128. This will keep up to 10k partitions open in the Table Readers instead of current 128
- optmise partition reload logic to not reload all the partitions every time a new partition is added/change in the table

This will speed up query execution for the tables with many partitions, like hourly partitions

### CODE REVIEW

File: core/src/main/java/io/questdb/cairo/TableReader.java [hunk 1]
```diff
@@ -144,7 +148,7 @@ public TableReader(
                 openPartitionInfo.setQuick(baseOffset, partitionTimestamp);
                 openPartitionInfo.setQuick(baseOffset + PARTITIONS_SLOT_OFFSET_SIZE, -1L); // -1L means it is not open
                 openPartitionInfo.setQuick(baseOffset + PARTITIONS_SLOT_OFFSET_NAME_TXN, txFile.getPartitionNameTxn(i));
-                openPartitionInfo.setQuick(baseOffset + PARTITIONS_SLOT_OFFSET_COLUMN_VERSION, columnVersionReader.getMaxPartitionVersion(partitionTimestamp));
+                openPartitionInfo.setQuick(baseOffset + PARTITIONS_SLOT_OFFSET_COLUMN_VERSION, -2);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/TableReader.java:151]: what does -2 mean and why?

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/TableReader.java:151]: It means it's not open, I can revert it here, it doesn't matter

Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableReader.java:151]: a comment or constant will do, don't need to revert, just making it easier to follow

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/TableReader.java:151]: added comment in [c50ecb8](https://github.com/questdb/questdb/pull/5010/commits/c50ecb8b8e01b475dd6d675d34eb96ad62977db1)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**